### PR TITLE
Fix the issue with adjoint-jacobian of adjoint ops

### PR DESCRIPTION
**Pull request by maliasadi**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
The Adjoint-Jacobian of adjoint operations in Lightning is incorrectly returning `0` if we don't decompose them. This issue stems from the ops serialization logic in Python, which mistakenly assumes `inverse = False` for all operations. This PR fixes the issue so that we can add `Adjoint(op)` for all supported `op` to the frozenset of devices operations that will also fix the blocker for merging PR #988.       

``` python

import pennylane as qml

dev = qml.device("lightning.qubit", wires=2)

@qml.qnode(dev, diff_method="adjoint")
def circuit(params):
    qml.adjoint(qml.RX(params[0], wires=[0]))
    qml.adjoint(qml.RY(params[0], wires=[0]))
    qml.adjoint(qml.RZ(params[0], wires=[0]))
    qml.adjoint(qml.SingleExcitation(params[0], wires=[0, 1]))
    return qml.expval(qml.Z(0))

params = qml.numpy.array([0.123])
qml.jacobian(circuit)(params)

```

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-78336]

### CODE REVIEW

**Review by maliasadi — COMMENTED** [review]
LGTM! @astralcai could you check your PR against the latest changes?

**Review by AmintorDusko — APPROVED** [review]
Looks good! Thanks!

File: pennylane_lightning/core/_serialize.py [hunk 1]
```diff
@@ -441,18 +444,20 @@ def get_wires(operation, single_op):
...
                 if isinstance(single_op, qml.QubitUnitary):
                     params.append([0.0])
                     mats.append(matrix(single_op))
-                elif not hasattr(self.sv_type, name):
-                    params.append([])
-                    mats.append(matrix(single_op))
                 else:
-                    params.append(single_op.parameters)
-                    mats.append([])
+                    if hasattr(self.sv_type, single_op.base.name if inverse else name):
```
Comment (astralcai) [inline pennylane_lightning/core/_serialize.py:455]: Why nesting the `if` inside the `else` now instead of chaining an `elif` like before?

**Review by astralcai — APPROVED** [review]
Looks good! Thank you for fixing this!

File: pennylane_lightning/core/_serialize.py (hunk 1 above)
Comment (maliasadi) [reply pennylane_lightning/core/_serialize.py:455]: This is mainly to reuse the logic for both inverse and named ops, while reducing the number of branches.

**Comment by maliasadi** [comment]
LGTM! Thanks @josephleekl for completing the PR :raised_hands:

**Pull request merged**